Tasks/Npm/npmtoolrunner.ts

```diff
@@ -23,13 +25,23 @@ export class NpmToolRunner extends tr.ToolRunner {
     public exec(options?: tr.IExecOptions): Q.Promise<number> {
         options = this._prepareNpmEnvironment(options) as tr.IExecOptions;
 
-        return super.exec(options);
+        return super.exec(options).catch((reason: any) => {
+            return this._printDebugLog(this._getDebugLogPath(options)).then((value: void): number => {
+                throw reason;
+            });
+        });
     }
 
     public execSync(options?: tr.IExecSyncOptions): tr.IExecSyncResult {
         options = this._prepareNpmEnvironment(options);
 
-        return super.execSync(options);
+        const execResult = super.execSync(options);
+        if (execResult.code !== 0) {
```

Comment:
This makes we wonder if there are any cases where npm emits a nonzero exit code, but does not emit a debug log. If that ever happened, you might print a stale debug log from another build.